nb-engine/nb-engine-core/src/main/java/io/nosqlbench/engine/api/activityimpl/uniform/AdapterResolver.java

```diff
@@ -0,0 +1,62 @@
+package io.nosqlbench.engine.api.activityimpl.uniform;
+
+/*
+ * Copyright (c) nosqlbench
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+
+import io.nosqlbench.adapter.diag.DriverAdapterLoader;
+import io.nosqlbench.adapters.api.activityimpl.uniform.DriverAdapter;
+import io.nosqlbench.adapters.api.activityimpl.uniform.Space;
+import io.nosqlbench.adapters.api.activityimpl.uniform.flowtypes.CycleOp;
+import io.nosqlbench.nb.annotations.ServiceSelector;
+import io.nosqlbench.nb.api.components.core.NBComponent;
+import io.nosqlbench.nb.api.config.standard.NBConfigModel;
+import io.nosqlbench.nb.api.config.standard.NBConfigurable;
+import io.nosqlbench.nb.api.config.standard.NBConfiguration;
+import io.nosqlbench.nb.api.errors.OpConfigError;
+import io.nosqlbench.nb.api.labels.NBLabels;
+
+import java.util.ServiceLoader;
+import java.util.function.BiFunction;
+
+public class AdapterResolver
+    //    implements BiFunction<String, NBConfiguration, DriverAdapter<? extends CycleOp<?>, Space>>
+{
+    public DriverAdapter<? extends CycleOp<?>, Space> apply(
+        NBComponent parent,
+        String name,
+        NBConfiguration configSuperset
+    )
+    {
+        ServiceSelector<DriverAdapterLoader>
+            loader =
+            ServiceSelector.of(name, ServiceLoader.load(DriverAdapterLoader.class));
+        DriverAdapterLoader
+            dal =
+            loader.get()
+                .orElseThrow(() -> new OpConfigError("No DriverAdapterLoader found for " + name));
+        DriverAdapter<CycleOp<?>, Space> adapter = dal.load(parent, NBLabels.forKV());
```

Comment:
Several comments here:
1. I don't like this code style with unneeded multiple lines.
2. Call the `loader` variable `selector`.
3. Call the `dal` variable `loader`.
4. Use a version of the "helpful" config error.
   ```java
    //        if (defaultDriverName.isPresent() && defaultAdapter.isEmpty()) {
    //            throw new BasicError("Unable to load '" + defaultDriverName.get() + "' driver adapter.\n" + "Rebuild NB5 to include this " + "driver adapter. Change" + " '<activeByDefault>false</activeByDefault>' for the driver in" + " './nb-adapters/pom.xml' and" + " './nb-adapters/nb-adapters-included/pom.xml' first.");
    //        }
   ```

Reply:
Fully agreed on 1-3.
For 1) I was experimenting with auto-formatting to address some of the inconsistencies and this was a format from an earlier setting which needs to be updated. I think we'll need settle on a formatting standard separately from this PR, and I'm eager for us to do that to ease PRs and reviews.

On 4) I'm ok with that prescriptive error as long as we don't send users down a false path. (i.e, they just had a typo, and no such adapter exists)

Reply:
Sure - we can approach 4) in other ways including documentation.